### PR TITLE
Use HookRelay to send webhooks

### PR DESCRIPTION
**Pull request by segiddins**
See https://www.hookrelay.dev for docs

This way, we can have nice things like retries & delivery tracking without building it ourselves

Next step will be recording more info on receiving reports in the DB, so we can automatically disable consistently failing webhooks

### CODE REVIEW

**Review by indirect — APPROVED** [review]

File: app/jobs/notify_web_hook_job.rb [hunk 1]
```diff
@@ -14,23 +17,26 @@ class NotifyWebHookJob < ApplicationJob
 
   attr_reader :webhook, :protocol, :host_with_port, :version, :rubygem
 
+  ERRORS = (HTTP_ERRORS + [RestClient::Exception, SocketError, SystemCallError, OpenSSL::SSL::SSLError]).freeze
+  retry_on(*ERRORS)
+
+  # has to come after the retry on
+  discard_on(RestClient::UnprocessableEntity) do |j, e|
```
Comment (simi) [inline app/jobs/notify_web_hook_job.rb:24]: Any reason to not keep it self-explanatory like `|job, error|`?

File: app/jobs/notify_web_hook_job.rb [hunk 2]
```diff
@@ -3,8 +3,11 @@ class NotifyWebHookJob < ApplicationJob
   include TraceTagger
 
   queue_as :default
+  queue_with_priority PRIORITIES.fetch(:web_hook)
```
Comment (simi) [inline app/jobs/notify_web_hook_job.rb:6]: :thinking: Seems it makes sense to move `PRIORITIES` from `delayed_job.rb` initializer into application.rb or something generic. I can do that in followup PR.

File: app/jobs/notify_web_hook_job.rb [hunk 3]
```diff
@@ -42,6 +48,47 @@ def authorization
     Digest::SHA2.hexdigest([rubygem.name, version.number, webhook.api_key].compact.join)
   end
 
+  def hook_relay_url
+    "https://api.hookrelay.dev/hooks/#{ENV['HOOK_RELAY_ACCOUNT_ID']}/#{ENV['HOOK_RELAY_HOOK_ID']}/webhook_id-#{webhook.id}"
+  end
+
+  def use_hook_relay?
+    # can't use hook relay for `perform_now` (aka an unenqueued job)
+    # because then we won't actually hit the webhook URL synchronously
```
Comment (simi) [inline app/jobs/notify_web_hook_job.rb:57]: Is that actual problem? Bypassing hookrelay makes sense in development environment for simple testing, but is there any reason to bypass that in production? Looking at the code, testing webhook (calling `/api/v1/web_hooks/fire`) is the only place using `perform_now`. Would it make any difference to call hookrelay in there as well? Any idea what's the delay?

**Review by simi — APPROVED** [review]
Overall looks good, I left some minor comments and one question. I think hookrelay setup and logic can be wrapped in simple class place in `lib` directory, but that could be improved later, no need to block this PR.

File: app/jobs/notify_web_hook_job.rb (hunk 3 above)
Comment (segiddins) [reply app/jobs/notify_web_hook_job.rb:57]: We use the return value from performing the hook in the response to the `fire` endpoint

Comment (simi) [reply app/jobs/notify_web_hook_job.rb:57]: OK, let's leave it as is than. Would be great to find out way to remove this later.

**Pull request merged**